lib/Sema/CSBindings.cpp

```diff
@@ -374,6 +375,10 @@ ConstraintSystem::getPotentialBindings(TypeVariableType *typeVar) {
         result.InvolvesTypeVariables = true;
       continue;
     }
+    
+    // Do not attempt to bind to ErrorType.
+    if (type->hasError())
```

Comment:
Do you have an example of where we would hit this?

I don't recall what I ran into when I originally added my assert.